### PR TITLE
Composable roles: Fix the node count when nodes fails introspection

### PR DESCRIPTION
**Pull request by asyedham**
This PR fixes
failed_nodes_machine_type fact
Adjusts node count for computes after introspection fails
Optimize the code to run faster with overcloud tag

### CODE REVIEW

File: post_introspect.yml [hunk 1]
```diff
@@ -53,8 +60,83 @@
...
-    - name: remove the templates dir
-      file:
-        state: absent
-        path: /home/stack/templates
-      ignore_errors: true
+    - name: setting node capabilities (boot_mode bios)
+      vars:
+        boot_mode: bios
+      include_tasks: tasks/set_boot_mode.yml
+      when: osp_release|int == 17 or composable_roles
```
Comment (sanjaychari) [inline post_introspect.yml]: I think `osp_release|int >= 17` can be checked instead?

File: tasks/set_root_hints.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@
   with_items:
     - "{{ total_nodes.stdout_lines }}"
 
-- name: set root hints based on model (740XD G-CL)
+- name: set root hints based on model
```
Comment (sanjaychari) [inline tasks/set_root_hints.yml:10]: Are we setting root hints for all models here? If that's the case, maybe `name: set root hints` is a better name?

File: post_introspect.yml (hunk 1 above)
Comment (asyedham) [reply post_introspect.yml]: Done

File: post_introspect.yml [hunk 3]
```diff
@@ -53,8 +60,83 @@
...
+      vars:
+        boot_mode: bios
+      include_tasks: tasks/set_boot_mode.yml
+      when: osp_release|int >= 17 or composable_roles
+
+    - name: get overcloud node images
+      vars:
+        image_path: /home/stack/images
+      include_tasks: tasks/overcloud_images.yaml
+      when: osp_release|int == 17
```
Comment (sanjaychari) [inline post_introspect.yml:73]: Can you change this also to  `when: osp_release|int >= 17`?

Comment (asyedham) [reply post_introspect.yml:73]: As we are facing the issue with overcloud images for only osp 17, the condition i have added make sense. Thoughts?

Comment (sanjaychari) [reply post_introspect.yml:73]: Yeah, makes sense.

**Comment by sanjaychari** [comment]
LGTM

**Pull request merged**